mesonbuild/modules/python.py

```diff
@@ -145,6 +147,7 @@ def __init__(self, python: 'PythonExternalProgram', interpreter: 'Interpreter'):
         })
 
     @permittedKwargs(mod_kwargs)
+    @FeatureNewKwargs('python_installation.extension_module', '1.3.0', ['limited_api'])
     @typed_pos_args('python.extension_module', str, varargs=(str, mesonlib.File, CustomTarget, CustomTargetIndex, GeneratedList, StructuredSources, ExtractedObjects, BuildTarget))
     @typed_kwargs('python.extension_module', *_MOD_KWARGS, _DEFAULTABLE_SUBDIR_KW, allow_unknown=True)
```

Comment:
I might be wrong, but I think you need to add `limited_api` as a typed kwarg.

```python
_PY_EXT_MOD_KWARGS = (
  *_MOD_KWARGS,
  # add your kwarg here.
)

Reply:
You can then drop the type checking you do later.

Reply:
And I don't think you need the FeatureNew you added then

Reply:
We don't "have" to, but as of a couple days ago we have typed kwargs in the first place here.

It's incomplete (allow_unknown=True) because the `shared_module()` function isn't complete but we should definitely start adding new things here, now, to take advantage of the improved type checking and builtin `since=` handling.

This would let us get rid of:
```python
        if not isinstance(limited_api_version, str):
            raise InvalidArguments('"limited_api" argument must be a string.')
```